pkg/apis/monitoring/v1/alertmanager_types.go

```diff
@@ -177,6 +177,8 @@ type AlertmanagerSpec struct {
 	// SecurityContext holds pod-level security attributes and common container settings.
 	// This defaults to the default PodSecurityContext.
 	SecurityContext *v1.PodSecurityContext `json:"securityContext,omitempty"`
+	DNSPolicy       DNSPolicy              `json:"dnsPolicy,omitempty"`
```

Comment:
(nit) optional fields need to be pointers per API conventions. We'd need to copy the description from prometheus_types.go too.
```suggestion
	DNSPolicy       *DNSPolicy              `json:"dnsPolicy,omitempty"`
```